### PR TITLE
Termination grace time and prestop drain

### PR DESCRIPTION
**Pull request by respringer**
Now the default TerminationGracePeriodSeconds is set to 2 minutes.  Also, by default, the "cassandra" container will have lifecycle prestop hook that calls node drain.  This call can handle both "insecure" and "MTLS" management API modes.

### CODE REVIEW

**Review by jimdickinson — CHANGES_REQUESTED** [review]
should `git rm operator/pkg/utils/test-jks`

File: tests/timeout_prestop_termination/timeout_prestop_termination_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+			step = "checking that the dc no longer exists"
+			json = "jsonpath={.items}"
+			k = kubectl.Get("CassandraDatacenter").
+				WithLabel(dcLabel).
+				FormatOutput(json)
+			ns.WaitForOutputAndLog(step, k, "[]", 300)
+
+			elapsedTime := startTime.Sub(time.Now())
+
+			Expect(elapsedTime.Minutes() <= 3).To(BeTrue(), "Stop should have completed in under 3 minutes")
```
Comment (jimdickinson) [inline tests/timeout_prestop_termination/timeout_prestop_termination_suite_test.go:98]: 👍

**Comment by respringer** [comment]
> should `git rm operator/pkg/utils/test-jks`

Arg, why does that file even get created?  I am usually pretty good at explicitly ignoring it, but we should either prevent it from being made or add it to the gitignore.  I will take a look.

**Comment by jimdickinson** [comment]
> > should `git rm operator/pkg/utils/test-jks`
> 
> Arg, why does that file even get created? I am usually pretty good at explicitly ignoring it, but we should either prevent it from being made or add it to the gitignore. I will take a look.

No worries! Yes feel free to add it to .gitignore - I hadn't seen this particular file in this location before

**Review by jimdickinson — APPROVED** [review]
looks very good!

File: tests/testdata/default-single-rack-single-node-prestop-dc.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    insecure: {}
+  size: 1
+  podTemplateSpec:
+    spec:
+      containers:
+        - name: "cassandra"
+          lifecycle:
+            preStop:
+              exec:
+                command: ["/bin/sleep", "6000s"]
```
Comment (respringer) [inline tests/testdata/default-single-rack-single-node-prestop-dc.yaml:19]: This is to force the shutdown to take forever so the grace period timeout will kick-in.

File: operator/pkg/utils/crypto_test.go [hunk 3]
```diff
@@ -79,6 +80,10 @@ func Test_GetJKS(t *testing.T) {
 	if len(jks) == 0 {
 		t.Errorf("JKS blob too small")
 	}
-	ioutil.WriteFile("test-jks", jks, 0644)
 
+	defer func() {
+		_ = os.Remove("test-jks")
```
Comment (sandoichi) [inline operator/pkg/utils/crypto_test.go:85]: :+1:

**Review by sandoichi — APPROVED** [review]
:100:

**Pull request merged**